src/store/Chat/index.js

```diff
@@ -161,9 +180,34 @@ function ChatReducer (state = initialState, action) {
       //   });
       // }
 
+      // We need to run through the entire array of messages and aggregate them
+      //  into similar sets
+
+      if (action.payload.length > 0) {
+        sortedPayload = action.payload.sort((curr, next) => (
+          new Date(curr.timestamp) - new Date(next.timestamp)
+        ));
+        // There should be an algorithm here that would speed things up
```

Comment:
is there any way that Go could handle this? 

edit: or at least if ever loading from a DB handle that part of it; and new socket messages get grouped accordingly?